navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/MapboxNavigation.java

```diff
@@ -438,7 +455,7 @@ public void onServiceConnected(ComponentName componentName, IBinder service) {
       if (alertLevelChangeListeners != null) {
         navigationService.setAlertLevelChangeListeners(alertLevelChangeListeners);
       }
-
+      progressChangeListeners.add(MapboxNavigation.this);
```

Comment:
The if null checks were leftovers from before when we didn't add the listeners to a list, I've removed them since the list are always set in the constructor. The MapboxNavigation objects listening in to get notified when the user arrives at the destination. When this occurs, the sessions, automatically ended. Otherwise, the service would continue running in the background even though the navigation sessions over. 

I'm not really happy with the current setup with everything being passed inside the onConnected callback since this is only called once when the service is first created, but I'll need more time to refactor.